### DISCUSSION TITLE
Rethrowing in error bounday - keep status code

### DISCUSSION THREAD

**Discussion by nphmuller** [remix-run/remix#6128] [Q&A]
Currently I have a couple of unhappy paths that have to be handled the same way across components.
There are also some unhappy paths that are for a specific component.

The way I currently handle this is that the errorboundary of the component rethrows, and the common unhappy paths are handled in the layout, with an <Oops /> error that's being rendered if it's really unknown what's going on.

The issue is as follows: the loader throws an error with the 400 code. The error boundary in the same component neatly returns a 400. However, if the error ends up in the layout error boundary, suddenly the status code is a 500.

Is there any way for me to allow `<UnhappyPathButKnownError />` in `root.tsx` return that same status code (400)?

Or should I just use my error boundaries in a completely different way?

component.tsx
```
export const loader = async({ params }: LoaderArgs) => {
  const response = await getSomething(params.id as string);
  if (response.error) {
    throw json(response, 400)
  }
  return json(response)
}

export function ErrorBoundary(): JSX.Element {
  const routeError = useRouteError()
  if (!isRouteErrorResponse(routeError)) throw routeError
  if (!isKnownError(routeError)) throw routeError

  <UnhappyPathForThisComponent error={routeError} />
}
```

root.tsx
```
export function ErrorBoundary(): JSX.Element {
  const routeError = useRouteError()
  if (!isRouteErrorResponse(routeError)) return <Oops />
  if (!isKnownError(routeError)) return <Oops />

  <UnhappyPathButKnownError error={routeError} />
}
```

**Comment by SupremeOverlordTheFirst** [comment]
@nphmuller did you ever work out how to do this? I have a similar scenario, curious if you ever managed to bubble the status code?
